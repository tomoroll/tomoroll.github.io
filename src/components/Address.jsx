import React from 'react';
import PropTypes from 'prop-types';
import { etherscanUrls } from '../utils/etheroll-contract';


function Address({ address, network }) {
  const url = `${etherscanUrls[network]}/address/${address}`;
  return <a href={url} className="text-monospace">{address}</a>;
}
Address.propTypes = {
  address: PropTypes.string.isRequired,
  network: PropTypes.number.isRequired,
};

export default Address;
