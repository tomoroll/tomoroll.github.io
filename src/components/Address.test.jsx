import React from 'react';
import renderer from 'react-test-renderer';
import Address from './Address';

it('renders correctly', () => {
  const tree = renderer
    .create(<Address address="0x0123" network={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
