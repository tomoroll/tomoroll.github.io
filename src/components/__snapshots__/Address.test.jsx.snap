// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="text-monospace"
  href="https://ropsten.etherscan.io/address/0x0123"
>
  0x0123
</a>
`;
